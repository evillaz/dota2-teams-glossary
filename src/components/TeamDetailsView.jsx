import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useLocation } from 'react-router-dom';
import { getTeamDetails } from '../redux/teams/teamsThunk';

const TeamDetailsView = () => {
  const path = useLocation();
  const teamId = Number(path.pathname.split('/')[2]);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getTeamDetails(teamId));
  }, [dispatch, teamId]);

  const { teamDetails, teams } = useSelector((store) => store.teams);

  const team = teams.find((t) => t.teamId === teamId);
  const handleImageError = (e) => {
    e.target.src = '/logo512.png';
    e.target.onerror = null;
  };

  return (
    <>
      <div className="detailsWrapper">
        {teamDetails.isLoading && (
          <span className="detailsLoading">
            Loading...
          </span>
        )}
        {(!teamDetails.isLoading && team) && (
          <>
            <div className="detailsHeader">
              <h2>
                {team.teamName}
              </h2>
              <img
                className="teamDetailsLogo"
                src={team.logo}
                alt="teamLogo"
                onErrorCapture={handleImageError}
              />
            </div>
            <table className="detailsTable">
              <thead className="detailsTableHeader">
                <tr>
                  <th>
                    Current Members
                  </th>
                </tr>
                <tr>
                  <th>
                    Name
                  </th>
                  <th>
                    Games
                  </th>
                  <th>
                    Winrate
                  </th>
                </tr>
              </thead>
              <tbody className="teamPlayers">
                {teamDetails.details.map((d) => (
                  <tr key={d.account_id}>
                    <td>{d.name}</td>
                    <td>{d.games_played}</td>
                    <td>{d.winrate}</td>
                  </tr>
                ))}
              </tbody>
            </table>

          </>
        )}

      </div>
    </>
  );
};

export default TeamDetailsView;
