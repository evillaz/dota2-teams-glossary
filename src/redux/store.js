import { configureStore } from '@reduxjs/toolkit';
import teamsReducer from './teams/teamSlice';

const store = configureStore({
  reducer: {
    teams: teamsReducer,
  },
});

export default store;
