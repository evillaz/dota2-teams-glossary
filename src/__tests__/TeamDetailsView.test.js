import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import store from '../redux/store';
import TeamDetailsView from '../components/TeamDetailsView';

describe('TeamDetailsView component', () => {
  it('should render correctly', () => {
    const fakeTeamDetails = {
      isLoading: false,
      details: [
        {
          account_id: 123,
          name: 'Player 1',
          games_played: 10,
          winrate: 50,
        },
        {
          account_id: 456,
          name: 'Player 2',
          games_played: 15,
          winrate: 60,
        },
      ],
    };

    const fakeTeams = [
      {
        teamId: 1,
        teamName: 'Team A',
        logo: 'logo.png',
      },
      {
        teamId: 2,
        teamName: 'Team B',
        logo: 'logo.png',
      },
    ];

    const { container } = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={[`/teams/${fakeTeams[0].teamId}`]}>
          <Routes>
            <Route path="/teams/:teamId" element={<TeamDetailsView />} />
          </Routes>
        </MemoryRouter>
      </Provider>,
    );

    store.dispatch({ type: 'teams/getTeamDetails/fulfilled', payload: fakeTeamDetails });
    store.dispatch({ type: 'teams/getTeams/fulfilled', payload: fakeTeams });

    expect(container).toMatchSnapshot();
  });
});
