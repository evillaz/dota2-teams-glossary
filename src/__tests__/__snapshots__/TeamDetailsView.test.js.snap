// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamDetailsView component should render correctly 1`] = `
<div>
  <div
    class="detailsWrapper"
  >
    <span
      class="detailsLoading"
    >
      Loading...
    </span>
  </div>
</div>
`;
